### ISSUE TITLE
Error on launch (POP_OS! 22.04)

### ISSUE THREAD

**Issue by Allan-Bruno** [flathub/com.spotify.Client#277]
output on terminal launch:
`allan@allan-notebook:~$ flatpak run com.spotify.Client
allan@allan-notebook:~$ /app/extra/bin/spotify: /usr/lib/x86_64-linux-gnu/libcurl.so.4: no version information available (required by /app/extra/bin/spotify)
[spotifywm] attached to spotify
/app/extra/share/spotify/spotify: /usr/lib/x86_64-linux-gnu/libcurl.so.4: no version information available (required by /app/extra/share/spotify/spotify)
/app/extra/share/spotify/spotify: /usr/lib/x86_64-linux-gnu/libcurl.so.4: no version information available (required by /app/extra/share/spotify/spotify)
[spotifywm] attached to spotify
[spotifywm] attached to spotify
`

**Comment by jeroen1602** [comment]
I have the same problem running Fedora linux 39 with KDE Plasma.

I thought it might be a wayland issue but it also won't launch when running with X11.

I have tried looking through `ps aux` and `top` to see if a spotify process was already running but I couldn't find one.

It also happens one a clean boot.

**Comment by Erick555** [comment]
Maybe try disabling [gpu cache](https://github.com/flathub/com.spotify.Client#making-flags-persistent).

**Comment by jeroen1602** [comment]
First I updated again to the latest stable version with hash: `b23c35bae652b4975189aa2f80fde9b52fa6356959f6562bebf0692eb1892bf1`

I tried creating the config file and disabling GPU cache. But no dice, it's still not launching.

I also tried running `flatpak run --command="bash" com.spotify.Client` and then run `spotify ----disable-gpu-shader-disk-cache --show-console` in the new bash instance. But that also didn't work.

**Comment by TingPing** [comment]
FWIW I tried reverting recent commits and it was still broken. Also disabling `spotifywm` didn't change anything.

**Comment by Allan-Bruno** [comment]
> None of this is an error. It found an existing window. Maybe it is minimized to the tray?

when I made the launch there were none processes of spotify, when launched, the process was created and stayed still for a minute, after that it killed itself

**Comment by Razer0123** [comment]
Same issue, i'm unable to run Spotify

**Comment by ErikMekiassen** [comment]
@jeroen1602 running
`sudo flatpak update --commit=d22ab2f6059cca52bc56754ca1860fb41b0d9c6333975bd8f2eb375fde5a4d84 com.spotify.Client`
worked for me as well

**Comment by gurnben** [comment]
> @jeroen1602 running `sudo flatpak update --commit=d22ab2f6059cca52bc56754ca1860fb41b0d9c6333975bd8f2eb375fde5a4d84 com.spotify.Client` worked for me as well

Kudos - this worked for me as well - hopefully a zero-intervention fix is out soon

**Comment by Erick555** [comment]
There isn't a single error message posted so there isn't a starting point to make a fix.

At first it would be nice to generalize on what kind of systems this happens and see what they have in common that triggers the issue (I have no issue myself) - everyone commenting should post at least distro version, desktop environment version and gpu version.

**Comment by ErikMekiassen** [comment]
Good point!
Fedora 38
gnome 44 
Wayland (not sure if thats self explanitory)
gpu: amd 6800xt

**Comment by Allan-Bruno** [comment]
Pop!_OS 22.04
Gnome 42.5
Xorg
HD graphics 520

**Comment by Allan-Bruno** [comment]
> @jeroen1602 running `sudo flatpak update --commit=d22ab2f6059cca52bc56754ca1860fb41b0d9c6333975bd8f2eb375fde5a4d84 com.spotify.Client` worked for me as well

for some reason it appers
`erro: com.spotify.Client not installed`

checked if the source was wrong

`allan@allan-notebook:~$ flatpak install flathub com.spotify.Client
Looking for matches…
Ignoring: com.spotify.Client/x86_64/stable is already installed
`

**Comment by ErikMekiassen** [comment]
@Allan-Bruno i got the same problem when i had to re update the commit but it  promted me to install it and then it worked again. not sure what flatpak update does exactly but i think its a flatpack problem

**Comment by Allan-Bruno** [comment]
> @Allan-Bruno i got the same problem when i had to re update the commit but it  promted me to install it and then it worked again. not sure what flatpak update does exactly but i think its a flatpack problem

how exactly you made it?

**Comment by ErikMekiassen** [comment]
Im not 100% sure but it happened after restart. spotify is the first thing i opened after firefox. i stil have the spotify icon but nothing happened. i ran flatpack run com.spotify.Client or something to get the prompt

**Comment by danielepusceddu** [comment]
> There isn't a single error message posted so there isn't a starting point to make a fix.
> 
> At first it would be nice to generalize on what kind of systems this happens and see what they have in common that triggers the issue (I have no issue myself) - everyone commenting should post at least distro version, desktop environment version and gpu version.

Fedora 39, Wayland Gnome 45.5, Intel UHD Graphics 620

**Comment by ErikMekiassen** [comment]
i do have an igpu in my intel i9 12900k so i might have  similar drivers to @Allan-Bruno @danielepusceddu

**Comment by ErikMekiassen** [comment]
running 
flatpak run com.spotify.Client
now gives this after restart (i updated to latest commit before the restart but it forgets it each time i reboot):
/app/extra/bin/spotify: /usr/lib/x86_64-linux-gnu/libcurl.so.4: no version information available (required by /app/extra/bin/spotify)
[emvit@10-13-37-63 ~]$ [spotifywm] attached to spotify
/app/extra/share/spotify/spotify: /usr/lib/x86_64-linux-gnu/libcurl.so.4: no version information available (required by /app/extra/share/spotify/spotify)
/app/extra/share/spotify/spotify: /usr/lib/x86_64-linux-gnu/libcurl.so.4: no version information available (required by /app/extra/share/spotify/spotify)
[spotifywm] attached to spotify
[spotifywm] attached to spotify

**Comment by Erick555** [comment]
All of those messages are fine.

**Comment by stefanfisk** [comment]
I was just affected by this as well after restarting my computer. `flatpak update --commit=d22ab2f6059cca52bc56754ca1860fb41b0d9c6333975bd8f2eb375fde5a4d84 com.spotify.Client` worked.

Pop!_OS 22.04 LTS
GNOME 42.9
Xorg
AMD RX580

**Comment by jeroen1602** [comment]
I might have found a clue.

I first launched a development shell inside of flatpak using `flatpak run --command=bash --devel com.spotify.Client`

I then ran the spotify binary (inside the flatpak) using strace `strace /app/extra/bin/spotify`.

In the output of strace I noticed that it would quit right after a call to `readLink` with the file `/home/jeroen/.var/app/com.spotify.Client/cache/spotify/SingletonSocket`.

Trying to read this symbolic link using `file /home/jeroen/.var/app/com.spotify.Client/cache/spotify/SingletonSocket` said it was a broken link.

After that I removed all the cache using `rm -rf /home/jeroen/.var/app/com.spotify.Client/cache/` after that I could start Spotify normally.

TL;DR

Removing the cache of spotify seemed to work. Which can be found at `~/.var/app/com.spotify.Client/cache/`

**Comment by stefanfisk** [comment]
Updating `com.spotify.Client` made it stop launching again. But `rm -rf ~/.var/app/com.spotify.Client/cache/` made it work!

**Comment by agriffit79** [comment]
`rm -rf ~/.var/app/com.spotify.Client/cache` fixes for me also.

**Comment by Allan-Bruno** [comment]
`rm -rf ~/.var/app/com.spotify.Client/cache` worked for me too, idk if the devs are seeing this issue, so I'll let it open until then, but in case anyone need help on this, just use the command

**Comment by TingPing** [comment]
Removing the cache doesn't resolve anything here. :/

**Comment by TingPing** [comment]
The last relevant strace logs I get are:

```
stat("/sys/fs/cgroup/cpuset/chrome", 0x7fffb99a4708) = -1 ENOENT (No such file or directory)
stat("/sys/fs/cgroup/schedtune/chrome", 0x7fffb99a4708) = -1 ENOENT (No such file or directory)
setpriority(PRIO_PROCESS, 0, -1)        = -1 EACCES (Permission denied)
```

**Comment by pavelzherdin** [comment]
I stumbled on the same issue. OpenSUSE Tumbleweed, gnome 46, Radeon 6950.

Removing the cache folder helped. I put aside the broken folder, but it's 3.1g.

I tried to figure out what exactly had to be removed, and in my case there were 3 broken links in
.var/com.spotify.Client/cache/spotify
Removing only those made it start again. They looked like this
```
lrwxrwxrwx   1 username username   20 Apr  6 21:27 SingletonCookie -> 15038546741398103137
lrwxrwxrwx   1 username username   23 Apr  6 21:27 SingletonLock -> localhost.localdomain-6
lrwxrwxrwx   1 username username   50 Apr  6 21:27 SingletonSocket -> /tmp/.org.chromium.Chromium.RtiORB/SingletonSocket
```

**Comment by Erick555** [comment]
Did you check they were invalid inside sandbox?

**Comment by pavelzherdin** [comment]
> Did you check they were invalid inside sandbox?

They are. 15038546741398103137 doesn't exist either way.

**Comment by Razer0123** [comment]
When will the fix be released? i still have the issue

**Comment by Erick555** [comment]
Never because it's local file corruption. Just read what do to above.

**Comment by TingPing** [comment]
Deleting those files doesn't solve anything here so... there must be more to it?

**Comment by pavelzherdin** [comment]
> Did you check they were invalid inside sandbox?

To add to this, it looks like the client creates those files on start and some of them are bad links intentionally. For me, it's enough to remove `SingletonLock` only for the app to start.
In the bad cache
`SingletonLock -> localhost.localdomain-6`
In the working cache
`SingletonLock -> DESKTOP-MEQFCF4-6`

I experimented a bit by creating that file myself with different targets, some of them indeed prevent the client from starting. Going to stop now, this rabbit hole is too deep for me.

**Comment by cch264** [comment]
Haven't been able to use Spotify in weeks. Updates did not fix the issue nor did uninstalling and reinstalling as user or system wide. The only solution was the fix above. Thank you!!!

**Comment by axelsimon** [comment]
> Never because it's local file corruption. Just read what do to above.

That's a bit rough on regular users, don't you think?

In practice, this means you are using Spotify desktop normally and then it just stops working. You did nothing wrong, it just stops working. This is the sort of stuff that makes people think "the computer has a mind of its own".

I think we need to do better at making things work for users. Can't we detect this situation and then delete the cache automatically?
Because the alternative is to rely on the users: 
1. knowing they can run the app from the CLI,
2. doing so and seeing the error message 
3. searching for it 
4. finding this thread and finally
5.  manually running `rm -rf ~/.var/app/com.spotify.Client/cache/`

**Comment by Erick555** [comment]
> Can't we detect this situation and then delete the cache automatically?

No, and that's the problem. We don't control spotify behavior and they don't even support linux officially.

**Comment by Erick555** [comment]
> Can't we detect this situation and then delete the cache automatically?

That would require understanding what is happening.

**Comment by axelsimon** [comment]
Fair point.
A `pgrep -af spotify` however returns nothing, which is telling. But yeah, it could be considered to be a bit brutal to delete the cache just because a pgrep returns nothing.

**Comment by Erick555** [comment]
I see that spotify creates those three `Singleton*` symlinks on startup and removes them on exit so maybe failing to remove them causes the issue on next startup when there are the old ones. `rm ~/.var/app/com.spotify.Client/cache/spotify/Singleton*` in pre startup may help then.

**Comment by SamuelWAnderson45** [comment]
> rm ~/.var/app/com.spotify.Client/cache/spotify/Singleton*

Deleting just those singleton files fixes it for me

**Comment by nutuxedo** [comment]
> @jeroen1602 running `sudo flatpak update --commit=d22ab2f6059cca52bc56754ca1860fb41b0d9c6333975bd8f2eb375fde5a4d84 com.spotify.Client` worked for me as well

this works for me, thanks!

**Comment by Razer0123** [comment]
Surprised this has not been fixed yet

**Comment by sagbobbit** [comment]
FWIW, I've got the same issue with the same "solution." openSUSE Tumbleweed + XFCE.

**Comment by RoyaltyInTraining** [comment]
Same here on Fedora 40. The issue returns every time after wiping the app's data. Sometimes within a week, other times within a day.

**Comment by e455a81e-d3ba-41a2-bc6d-7aafb1d9a5cd** [comment]
I just encountered this issue as well on Fedora Kinoite _40.20240506.0_ with Spotify Version _1.2.31.1205.g4d59ad7c_.
Could this issue be related to forced closings of the Spotify application? When I rebooted today KDE asked me whether I wanted to logout the session even if some apps are still running. This could have resulted in a race condition that might have prevented Spotify from cleaning up the files in `~/.var/app/com.spotify.Client/cache/spotify/`.

**Comment by Erick555** [comment]
I tested it by killing spotify process - old `Singleton*` symlinks were left but I started spotify without issues.

**Comment by e455a81e-d3ba-41a2-bc6d-7aafb1d9a5cd** [comment]
I tried to reproduce it as well with a reboot between killing the spotify process unfortunately I hadn't had any success. I also looked at the [chromium code](https://chromium.googlesource.com/chromium/src/+/master/chrome/browser/process_singleton_posix.cc) that handles these singleton files and attempted to mess with the Singleton symlinks to get it to break, but it just works no matter how hard I try to break it.

**Comment by Onihani** [comment]
> I might have found a clue.
> 
> I first launched a development shell inside of flatpak using `flatpak run --command=bash --devel com.spotify.Client`
> 
> I then ran the spotify binary (inside the flatpak) using strace `strace /app/extra/bin/spotify`.
> 
> In the output of strace I noticed that it would quit right after a call to `readLink` with the file `/home/jeroen/.var/app/com.spotify.Client/cache/spotify/SingletonSocket`.
> 
> Trying to read this symbolic link using `file /home/jeroen/.var/app/com.spotify.Client/cache/spotify/SingletonSocket` said it was a broken link.
> 
> After that I removed all the cache using `rm -rf /home/jeroen/.var/app/com.spotify.Client/cache/` after that I could start Spotify normally.
> 
> TL;DR
> 
> Removing the cache of spotify seemed to work. Which can be found at `~/.var/app/com.spotify.Client/cache/`

This worked for me

**Comment by Razer0123** [comment]
Still having the same issue to this day

**Comment by Erick555** [comment]
@Razer0123 do you mean you remove Singleton* files every day and this issue re-occurs?

**Comment by Razer0123** [comment]
> @Razer0123 do you mean you remove Singleton* or entire cache files every day and this issue re-occurs?

Yes, it stops working, i delete either one of them, i am able to open the app, the next day (or so) it wont't work again

**Comment by Erick555** [comment]
That's extreme, what distro do you use?

**Comment by Razer0123** [comment]
> That's extreme, what distro do you use?

Latest Fedora version + Spotify flatpak

**Comment by Erick555** [comment]
What's the `uname -n` command output? Is something unusual in your system setup like dynamic hostname generation?

**Comment by Razer0123** [comment]
> What's the `uname -n` command output? Is something unusual in your system setup like dynamic hostname generation?

Is this what you are referring to?

`Linux host-xxxxxm 6.9.6-200.fc40.x86_64 #1 SMP PREEMPT_DYNAMIC Fri Jun 21 15:48:21 UTC 2024 x86_64 GNU/Linux
`
I just noticed it got fixed after changing DNS

**Comment by Razer0123** [comment]
Still having this issue :(

**Comment by mfau** [comment]
This just happened to me on Bazzite.
Deleting the cache folder worked.

**Comment by Razer0123** [comment]
I gave up and installed Spotify via RPM

**Comment by Nick-Heyart** [comment]
Occasionally having this issue on Fedora 40, fixed by deleting cache. Installing Spotify with LPF seems to have this bug as well.

**Comment by engin-karakurt** [comment]
Just had this issue on Fedora, deleting the cache at `~/.var/app/com.spotify.Client/cache/` like the other answers suggested works for me.

System info:

```
             .',;::::;,'. 
         .';:cccccccccccc:;,. 
      .;cccccccccccccccccccccc;.          OS: Fedora Linux 40 (KDE Plasma) x86_64
    .:cccccccccccccccccccccccccc:.        Host: B560M DS3H
  .;ccccccccccccc;.:dddl:.;ccccccc;.      Kernel: Linux 6.10.6-200.fc40.x86_64
 .:ccccccccccccc;OWMKOOXMWd;ccccccc:.     Uptime: 1 hour, 46 mins
.:ccccccccccccc;KMMc;cc;xMMc;ccccccc:.    Packages: 2692 (rpm), 3 (flatpak)
,cccccccccccccc;MMM.;cc;;WW:;cccccccc,    Shell: pk-command-not-found
:cccccccccccccc;MMM.;cccccccccccccccc:    Display (F24G3xTF): 1920x1080 @ 144 Hz in 24″ [External] *
:ccccccc;oxOOOo;MMM000k.;cccccccccccc:    Display (F24G3xTF): 1920x1080 @ 144 Hz in 24″ [External]
cccccc;0MMKxdd:;MMMkddc.;cccccccccccc;    DE: KDE Plasma 6.1.4
ccccc;XMO';cccc;MMM.;cccccccccccccccc'    WM: KWin (Wayland)
ccccc;MMo;ccccc;MMW.;ccccccccccccccc;     WM Theme: Breeze
ccccc;0MNc.ccc.xMMd;ccccccccccccccc;      Theme: Breeze (Dark) [Qt], Breeze [GTK3]
cccccc;dNMWXXXWM0:;cccccccccccccc:,       Icons: breeze-dark [Qt], breeze-dark [GTK3/4]
cccccccc;.:odl:.;cccccccccccccc:,.        Font: Noto Sans (10pt) [Qt], Noto Sans (10pt) [GTK3/4]
ccccccccccccccccccccccccccccc:'.          Cursor: breeze (24px)
:ccccccccccccccccccccccc:;,..             Terminal: konsole 24.8.0
 ':cccccccccccccccc::;,.                  CPU: 11th Gen Intel(R) Core(TM) i5-11400F (12) @ 4,40 GHz
                                          GPU: NVIDIA GeForce RTX 3060 [Discrete]
                                          Memory: 7,03 GiB / 15,48 GiB (45%)
                                          Swap: 398,50 MiB / 8,00 GiB (5%)
                                          Disk (/): 78,75 GiB / 673,83 GiB (12%) - btrfs
                                          Locale: de_DE.UTF-8
```

**Comment by Razer0123** [comment]
> Here what solved for me was:
> 
> cd $HOME/.var/app/com.spotify.Client rm -rf cache config flatpak override --user --socket=wayland com.spotify.Client
> 
> ![image](https://private-user-images.githubusercontent.com/20565821/374268567-23c79929-fb98-4f8e-83fd-2ee9515b2c8a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkwNzgzMjgsIm5iZiI6MTcyOTA3ODAyOCwicGF0aCI6Ii8yMDU2NTgyMS8zNzQyNjg1NjctMjNjNzk5MjktZmI5OC00ZjhlLTgzZmQtMmVlOTUxNWIyYzhhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDE2VDExMjcwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI5YTE2Y2U3NGY4ZjIwYTVkMTM1MTUxOGVhOGZmYTQ1MTFmOGQwNGI2ODMxNTE3NjczOGRjY2Q3Yzk4ZTkwMjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.cZPSFyR5tiohm_NHm3GYFae7IGR2lMLoUE5GWUoK0ig)

Why is my Spotify in a blueish window not using system theming after i have used this command?

**Comment by Erick555** [comment]
spotify flatpak automatically use wayland when wayland socket is enabled: https://github.com/flathub/com.spotify.Client/commit/5248553d4a3c01dce38929c108307cb7a30ed42a

It's hardcoded so you need to disable wayland socket again.

**Comment by elogikalaf** [comment]
after these steps I got it working 
sudo flatpak update --commit=d22ab2f6059cca52bc56754ca1860fb41b0d9c6333975bd8f2eb375fde5a4d84 com.spotify.Client
cd $HOME/.var/app/com.spotify.Client
rm -rf cache config
flatpak override --user --socket=wayland com.spotify.Client



question, should I stop upgrading spotify now? (e.g: using flatpak update)

**Comment by TeaDrinkingProgrammer** [comment]
I got it working using just the flatpak override command, not the downgrade. One thing I also found is that it works when first booting into it, but not when I try again another time.

I'm running Fedora 41

**Comment by xplosionmind** [comment]
Both with the wayland socket enabled and disabled, and by deleting `~/.var/app/com.spotify.Client/cache/` every time before running, the app does not even launch for me.

I cannot even read any debug log! Running `flatpak run com.spotify.Client --debug` fails without returning *anything*.

**Comment by RoyaltyInTraining** [comment]
Seems like the issue has fixed itself on my system (Fedora 41 KDE). It was really bad for me back in march, but in the last two months the app hasn't failed to open a single time. Does anyone still have this problem?

**Comment by tungstengmd** [comment]
clearing cache should work

**Comment by Seba244c** [comment]
Sometimes spotify just decides to not open, and it is stuck that way until i delete the cache.

**Comment by Heatsreef** [comment]
Issue still persists, would love a natively implemented fix

**Comment by jgrnrt** [comment]
Removing SingletonSocket, SingletonLock and SingletonCookie (in .var/app/com.spotify.Client/cache/spotify/) when the problem occurs also fixes the issue.

**Comment by totow93** [comment]
I had this problem multiple times a week i made me a script to delete the files on Login.
```
#!/bin/bash
find $HOME/.var/app/com.spotify.Client/cache/spotify/ -name 'Singleton*' -delete
```

**Comment by Hobbes118** [comment]
@jeroen1602 just worked for me a year later after bazzite update messed with spotify, your a lifesaver.

**Comment by alxns7** [comment]
Deleting the cache files fixed it for me. I think it happened because I ran a `reboot` command with spotify still open.

**Comment by tabney00** [comment]
Is the consensus then that Spotify MUST be closed manually in order to gracefully exit? This behavior has been annoying but I think it occurs when I restart the computer without closing Spotify, which is sometimes unavoidable. Will have to test this again, but it would make sense that if those singleton files aren't cleaned up properly they "correctly" prevent another instance from starting.
